cassiopeia/solver/MaxCutGreedySolver.py

```diff
@@ -92,33 +90,56 @@ def perform_split(
         Returns:
             A tuple of lists, representing the left and right partitions
         """
-        freq = 0
-        char = 0
-        state = ""
-        for i in mutation_frequencies:
-            for j in mutation_frequencies[i]:
-                if j != self.missing_char and j != "0":
+        best_frequency = 0
+        chosen_character = 0
+        chosen_state = ""
+        for character in mutation_frequencies:
+            for state in mutation_frequencies[character]:
+                if state != self.missing_char and state != "0":
                     # Avoid splitting on mutations shared by all samples
                     if (
-                        mutation_frequencies[i][j] > freq
-                        and mutation_frequencies[i][j]
+                        mutation_frequencies[character][state]
                         < len(samples)
-                        - mutation_frequencies[i][self.missing_char]
+                        - mutation_frequencies[character][self.missing_char]
                     ):
-                        char, state = i, j
-                        freq = mutation_frequencies[i][j]
-
-        if state == "":
+                        if self.priors:
+                            if (
+                                mutation_frequencies[character][state]
+                                * self.priors[character][state]
+                                > best_frequency
+                            ):
+                                chosen_character, chosen_state = (
+                                    character,
+                                    state,
+                                )
+                                best_frequency = (
+                                    mutation_frequencies[character][state]
+                                    * self.priors[character][state]
+                                )
+                        else:
+                            if (
+                                mutation_frequencies[character][state]
+                                > best_frequency
+                            ):
+                                chosen_character, chosen_state = (
+                                    character,
+                                    state,
+                                )
+                                best_frequency = mutation_frequencies[
+                                    character
+                                ][state]
```

Comment:
This is incorrect. We multiply by the _log_ of these priors. Please refer back to the original Cassiopeia GreedySolver for the correct math.

If I understand this, you'd like to weight the mutation frequency by the prior. In this case, you might want to do something like `-nlog(p)` like we did in the original paper.

Reply:
Hi Matt,
So I think this is a discrepancy in the way that we think about these weights.

I think that potentially the fact that I have labeled the weights as priors here is confusing. But the way that I see it, the algorithms should be able to take in generic weights by which to weight the frequencies. These weights can be derived from priors, in which case a dictionary of -np.log(p) weights could be used, or some other transformation on p. But I can also imagine using some other types of weights in the future. What do you think? This would require some preprocessing to create the dictionary of weights, an example of which you can see in the vanillagreedy_test.py testing file. What do you think? Is this reasonable?

Reply:
Hey Richard,

These are fair points, but we've sort of developed our inference framework in the context of priors being non-logged. As you know, the original Cassiopeia ms & codebase logged these priors when incorporating them into the Greedy solver. For any user using Cassiopeia right now, that's how they're accustomed to using the priors. Plus, in the Cassiopeia codebase we have functionality for calculating the priors as frequencies and then normalizing the probabilities when creating the character matrices. The most straightforward solution here, in my opinion, would be to have user specify the priors as usual and we do the log transformation on the backend as usual.  

An alternative would be to have a `priors` and `weights`. If only priors are specified, we set `self.weights` to be the log of the priors. Else, we take whatever the user passes as weights.

Thoughts?